ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,188 @@
+        in_feature: FeaturesSpecification,
+        out_feature: FeaturesSpecification,
+        mode: Literal["standard", "timewise", "monthly", "total"] = "standard",
+        pixelwise: bool = False,
+    ):
+        """
+        :param in_feature: The input feature to compute the T-Digest representation for.
+        :param out_feature: The output feature where to save the T-Digest representation of the chosen feature.
+        :param mode: The mode to apply to the timestamps and bands.
+        - The 'standard' mode computes the T-Digest representation \
+            for each band accumulating timestamps.
+        - The 'timewise' mode computes the T-Digest representation \
+            for each band and timestamp of the chosen feature.
+        - The 'monthly' mode computes the T-Digest representation \
+            for each band accumulating the timestamps per month.
+        - The 'total' mode computes the total T-Digest representation \
+            of the whole feature accumulating all timestamps, bands and pixels \
+                - cannot be used with pixelwise=True.
+        :param pixelwise: Decider whether to compute the T-Digest representation accumulating pixels or per pixel. \
+            Cannot be used with mode='total'.
+        """
+
+        self.mode = mode
+
+        self.pixelwise = pixelwise
+
+        if self.pixelwise and self.mode == "total":
+            raise ValueError("Total mode does not support pixelwise=True.")
+
+        self.in_feature = self.parse_features(in_feature, allowed_feature_types=partial(_is_input_ftype, mode=mode))
+        self.out_feature = self.parse_features(
+            out_feature, allowed_feature_types=partial(_is_output_ftype, mode=mode, pixelwise=pixelwise)
+        )
+
+        if len(self.in_feature) != len(self.out_feature):
+            raise ValueError(
+                f"The number of input ({len(self.in_feature)}) and output features ({len(self.out_feature)}) must"
+                " match."
+            )
+
+    def execute(self, eopatch: EOPatch) -> EOPatch:
+        """
+        Execute method that computes the TDigest of the chosen features.
+
+        :param eopatch: EOPatch which the chosen input feature already exists
+        """
+
+        if self.mode == "standard":
+            _process_standard(
+                in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch, pixelwise=self.pixelwise
+            )
+
+        elif self.mode == "timewise":
+            _process_timewise(
+                in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch, pixelwise=self.pixelwise
+            )
+
+        elif self.mode == "monthly":
+            _process_monthly(
+                in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch, pixelwise=self.pixelwise
+            )
+
+        elif self.mode == "total":
+            _process_total(in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch)
```

Comment:
since all of these methods use the `_looper` (except for the final one, but we can just ignore the shape there), let's extract the `for loop` outside, so that we'll have something like:
```python
        for in_feature_, out_feature_, shape in _looper(in_feature=in_feature, out_feature=out_feature, eopatch=eopatch):
            if self.mode == "standard":
                _process_standard(...)
            elif self.mode == "timewise":
                ...
```

Then you might also notice that the `if`s are kinda repetitive. And you can also make these functions return the `np.ndarray` instead of assigning the value in the function. So maybe we can do something like:
```python
processing_function =  {
    "timewise": _process_timewise,
    ...
}[self.mode]

for in_feature_, out_feature_, shape in _looper(in_feature=in_feature, out_feature=out_feature, eopatch=eopatch):
    eopatch[out_feature] = processing_function(in_feature, eopatch)
```

You'll need a special case for `monthly` but that can probably be solved with the use of `kwargs`.

Reply:
Please have a look at the [commit](https://github.com/sentinel-hub/eo-learn/pull/603/commits/21251363dc8c3f7254d788f99cd9d9c4c0981148) if it's okay like this.

Reply:
looks like it's going in the right direction for sure :D
I'll run our CI, it `pylint` will likely complain about certain names.

Besides that I also noticed that you always do `.flatten()` before passing arrays into `get_tdigest`, but you already flatten inside, so you don't need to do that.

While i'm sure we could reduce the code a bit further, I feel like it's in a good enough place for now.
After satistfying the CI and fixing the `flatten` thing, I would focus on setting up test cases.

I'd start with a small easy test case or two, just so we know the task runs and does its thing.

Reply:
Sry, completely missed the flatten-thing. Done now. Please confirm that you are taking care of the pylint and the first tests.

Reply:
ah, since you changed the name of the file, you also need to adjust the `__init__` file.

Reply:
ah, ups, dumb mistake. Done.

Reply:
Regarding the tests: what should they show? Which cases should be captured? How much of effort do you expect that to be?

Reply:
at the very least it would be nice to have one or two cases of things going as they should (just to check that the end-to-end process works). We sometimes just run it on some inputs and then assert that the statistics match the expected ones (see [here](https://github.com/sentinel-hub/eo-learn/blob/master/features/eolearn/tests/test_haralick.py) for instance).

You can also verify just that the object you receive is populated with correct types and that the shapes are ok (like [here](https://github.com/sentinel-hub/eo-learn/blob/master/features/eolearn/tests/test_feature_manipulation.py#L261)).

From my viewpoint we can also add the tests in a later MR if a deadline has to be met.

Reply:
I let a student quickly test the current version and it seems to work in the context of the eo-learn examples, we provide.
Hence, I would suggest, we continue without tests at a first stage as we have to finish the IGARSS project together which will be bit of work as well.
Later, these tests should definitely be added.

It would be great if you could take care of that and inform me what's been wrong so I can improve next time!